core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -98,11 +99,25 @@ class KafkaApis(val requestChannel: RequestChannel,
       request.apiLocalCompleteTimeMs = SystemTime.milliseconds
   }
 
+  //TODO: this method should only use request.header after all the requests are migrated to use client java request class.
```

Comment:
Agreed: minor, but it took me a bit to sort out what was going on here. Although part of that is due to fact that the RequestChannel.Request fields are kind of confusing to keep track of wrt which ones are non-null for which type of requests.

I think just changing this to an Option[Short] and maybe just adding short comments to the `case` lines to help clarify when we get it from the header (new Java-style requests) vs when we get it passed in (old Scala-style requests) would make this a lot clearer.